app/routes/activity.js

```diff
@@ -14,5 +14,16 @@ export default Ember.Route.extend({
 
   afterModel(activity) {
     this.set('activityManager.current', activity);
+  },
+
+  actions: {
+  	restart(){
+  		this.get('activityManager.current').setProperties({
+  			completedIndex: 0,
+  			error: null
+  		});
```

Comment:
We could add a `restart` method in the `activityManager` service so we hide the restart details from the `activity` route, what do you think?